### PR TITLE
tableToMarkdown fix - dictionary with special character

### PR DESCRIPTION
**Pull request by IkaDemisto**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17450

## Description
In case of passing dictionary with unicode key and list that includes non ascii character (for example: {'header_1': u'foo', 'header_2': [u'â']}) to `tableToMarkdown`, `UnicodeDecodeError` error was raised. The issue was due to joining unicode and str types.

## Required version of Demisto
4.5.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [x] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] All tests

### CODE REVIEW

**Comment by Itay4** [comment]
what in case non-ascii in table headers?

**Comment by IkaDemisto** [comment]
@Itay4 correct, fails in particular cases. Fails with `{'header_1': u'foo'}` but not with `{'header_1': [u'foo']}`. Will include fix for it also.

File: Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -830,9 +830,16 @@ def tableToMarkdown(name, t, headers=None, headerTransform=None, removeNull=Fals
...
+                try:
+                    mdResult += vals[0]
+                except UnicodeDecodeError:
+                    mdResult += vals[0].encode('utf-8')
             else:
-                mdResult += '|'.join(vals)
+                try:
+                    mdResult += '|'.join(vals)
+                except UnicodeDecodeError:
+                    vals = [str(v) for v in vals]
```
Comment (yaakovi) [inline Scripts/CommonServerPython/CommonServerPython.py]: let's use encode here too

File: Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -830,9 +830,16 @@ def tableToMarkdown(name, t, headers=None, headerTransform=None, removeNull=Fals
                                    True, True) for h in headers]
             mdResult += '|'
             if len(vals) == 1:
```
Comment (yaakovi) [inline Scripts/CommonServerPython/CommonServerPython.py]: let's remove the condition, the else part works for both cases

**Comment by anara123** [comment]
@IkaDemisto @yaakovi what is the status here?

**Comment by IkaDemisto** [comment]
Waited to content checkout. Will be force merged tomorrow @yaakovi?

**Comment by yaakovi** [comment]
changing `CommonServerPython` - will fail on tests.
unit tests and file validation passed

**Pull request merged**